### PR TITLE
CON-191, CON-190 64-bit numner support

### PR DESCRIPTION
**Pull request by samuelraeburn**
Docs:
Screenshot from 2021-07-02 11-54-15

### CODE REVIEW

**Comment by rkorte** [comment]
I have completely forgotten how to enter a comment on a specific line in a changed file. 

Anyway, I decided to make my edits in your branch and have you review them, if that's OK. A couple notes on my edits:

- I changed "that" to "which," which I _think_ is correct. I believe you mean that "Connector relies on a framework that only contains a single number type" (and, crucially, that number type is an IEEE-754 floating-point number). You do _not_ mean (I think) that Connector relies on a framework that contains numerous number types, and only one of them is an IEEE-754 floating-point number (which is the meaning if you use "that").
- I changed "we" to "Connector" to be more formal and specific; however, if the subject of that sentence isn't "Connector" but something else and you don't want to get into all that, you can change it back to "we." But if the subject is, say, "RTI Connext DDS," then you should say so (i.e., say who/what does "not use JavaScript's BigInt representation").

**Review by rkorte — CHANGES_REQUESTED** [review]
I added a commit with my suggested changes.

**Comment by rkorte** [comment]
UPDATE: I figured out how to enter comments. I had to click Add Review first.

**Review by rkorte — APPROVED** [review]

File: docs/data.rst [hunk 1]
```diff
@@ -191,17 +179,63 @@ getter: :meth:`SampleIterator.getNumber()`, :meth:`SampleIterator.getBoolean()`,
...
+than ``Number.MAX_SAFE_INTEGER`` (or smaller than ``Number.MIN_SAFE_INTEGER``),
+then you must take the following into account.
+
+64-bit values with an absolute value greater or equal to 2^53 can be set via:
+ - The type-agnostic setter, :meth:`Instance.set`. The values must be supplied
+   as strings, e.g., ``the_output.instance.set('my_uint64', '18446744073709551615')``.
+ - :meth:`Instance.setString`, e.g., ``the_output.instance.setString('my_uint64', '18446744073709551615')``.
+ - :meth:`Instance.setFromJson`, if the values are provided as strings, e.g., ``the_output.instance.setFromJson({my_uint64: '18446744073709551615'})``.
+
+64-bit values with an absolute value greater than 2^53 can be retrieved via:
```
Comment (alexcamposruiz) [inline docs/data.rst:211]: "greater" or "greater or equal"?

File: docs/data.rst [hunk 2]
```diff
@@ -191,17 +179,63 @@ getter: :meth:`SampleIterator.getNumber()`, :meth:`SampleIterator.getBoolean()`,
...
     a number, not a string.
 
+Accessing 64-bit integers
+^^^^^^^^^^^^^^^^^^^^^^^^^
+Internally, *Connector* relies on a framework that only contains a single number
+type, an IEEE-754 floating-point number. Additionally, *Connector* does not use 
+JavaScript's BigInt representation for numbers, meaning JavaScript has this same limitation.
+As a result, not all 64-bit integers can be represented with exact precision using all
+APIs.
+If your type contains uint64 or int64 members, and you expect them to be larger
```
Comment (alexcamposruiz) [inline docs/data.rst:201]: stylize int64 and uint64 (``)

File: docs/data.rst (hunk 1 above)
Comment (alexcamposruiz) [reply docs/data.rst:211]: I see the note below now.

**Review by alexcamposruiz — APPROVED** [review]

**Pull request merged**